### PR TITLE
Add duration unit format

### PR DESCRIPTION
**Pull request by cameronperera**
NOTE: I did upgrade webpack in this branch to verify that this fixed the build issues and it... DOES! Once this PR is merged, I will get the webpack PR updated and verify it is working.

# Description

This PR is replacing `intl-unofficial-duration-unit-format` with `@formatjs/intl-durationformat`. This is to help us get the [webpack upgrade](https://github.com/hashicorp/boundary-ui/security/dependabot/158) ([PR](https://github.com/hashicorp/boundary-ui/pull/2466)) into a working state.

The issue with `intl-unofficial-duration-unit-format` is due to an [export path](https://github.com/piuccio/intl-unofficial-duration-unit-format/issues/33) format in the released versions. Instead of patching this fix, we took this time to replace this package with the newer `@formatjs/intl-durationformat` polyfill.

## Screenshots (if appropriate)

One small difference between these packages is the polyfill ([spec for Int.DurationFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat/DurationFormat)) does not seem to have the same ability to define the format we want. Before if the session had a `max_duration` greater than 24 hours, it would keep adding to the hours and not show days. The new polyfill will display days now.
Before: `24:01:26 remaining`
After: `1 day, 0:01:26 remaining`

## How to Test

Use Vercel or an instance with session recordings.
In Admin UI, visit the session recordings pages and look for the different places we display the duration of the session recording.
In the Desktop Client, start a session and you should see "xx:xx:xx remaining" when viewing the session details.

## Checklist

- [ ] ~I have added before and after screenshots for UI changes~
- [ ] ~I have added JSON response output for API changes~
- [x] I have added steps to reproduce and test for bug fixes in the description
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by lisbet-alvarez — APPROVED** [review]
Looks great! 🙌

File: addons/core/addon/helpers/time-remaining.js [hunk 1]
```diff
@@ -29,14 +30,13 @@ export default class extends Helper {
       difference > 0 ? Math.floor(difference / 1000) : 0;
 
     // Format the duration
-    const duration = new DurationUnitFormat(this.intl.primaryLocale, {
-      style: 'timer',
-      format: `{hours}:{minutes}:{seconds} ${this.intl.t(
```
Comment (ZedLi) [inline addons/core/addon/helpers/time-remaining.js:34]: Hmm I forget, was there a reason we only limited it to hours before? I kind of like having more than hours

File: addons/core/tests/integration/helpers/time-remaining-test.js [hunk 2]
```diff
@@ -13,19 +13,26 @@ module('Integration | Helper | time-remaining', function (hooks) {
...
     this.set('expirationTime', new Date(Date.now() + 1000 * 60 * 60 * 24));
 
     await render(hbs`{{time-remaining this.expirationTime}}`);
 
     assert.strictEqual(this.element.textContent.trim(), '23:59:59 remaining');
   });
 
-  test('it handles negative time remaining', async function (assert) {
-    assert.expect(1);
+  test('it calculates the correct time remaining with more than 24 hours', async function (assert) {
```
Comment (ZedLi) [inline addons/core/tests/integration/helpers/time-remaining-test.js:24]: Could we also have a test for more than a week to see how it's formatted?

File: addons/core/tests/unit/utils/seconds-to-duration-test.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import { secondsToDuration } from 'core/utils/seconds-to-duration';
+import { module, test } from 'qunit';
+
+module('Unit | Utility | seconds-to-duration', function () {
+  test('it handles 0 correctly', function (assert) {
```
Comment (ZedLi) [inline addons/core/tests/unit/utils/seconds-to-duration-test.js:10]: Can we add a test of it handling a negative number correctly?

File: addons/core/addon/helpers/time-remaining.js (hunk 1 above)
Comment (cameronperera) [reply addons/core/addon/helpers/time-remaining.js:34]: I don't remember the reasoning but I think the new format allows for easier readability.

File: addons/core/tests/integration/helpers/time-remaining-test.js (hunk 2 above)
Comment (cameronperera) [reply addons/core/tests/integration/helpers/time-remaining-test.js:24]: Done.

File: addons/core/tests/unit/utils/seconds-to-duration-test.js (hunk 3 above)
Comment (cameronperera) [reply addons/core/tests/unit/utils/seconds-to-duration-test.js:10]: Done.

File: addons/core/addon/helpers/time-remaining.js (hunk 1 above)
Comment (calcaide) [reply addons/core/addon/helpers/time-remaining.js:34]: I remember a discussion, I do not remember the reason for it.

**Review by calcaide — APPROVED** [review]
Thanks for the work!

**Review by ZedLi — APPROVED** [review]
Looks good!
